implementation.js

```diff
@@ -114,13 +116,10 @@ if ('Map' in global) {
 var usingIterator = function (items) {
 	if (has(items, iteratorSymbol)) {
 		return items[iteratorSymbol]();
-	} else if ('Set' in global && 'Map' in global && isCallable(items.entries) && isCallable(items.values)) {
-		if (isSet(items)) {
-			return items.values();
-		}
-		if (isMap(items)) {
-			return items.entries();
-		}
+	} else if (hasMap && isMap(items)) {
+		return items.entries();
+	} else if (hasSet && isSet(items)) {
+		return items.values();
```

Comment:
and here, `values.call(items)` where `values` is `var values = hasSet && Set.prototype.values;`